components/automate-cli/cmd/chef-automate/pullAndGenerateConfig.go

```diff
@@ -192,14 +192,14 @@ type HATfvars struct {
 	SecretsStoreFile              string   `json:"secrets_store_file"`
 }
 type PullConfigs interface {
-	pullOpensearchConfigs() (map[string]*ConfigKeys, error)
-	pullPGConfigs() (map[string]*ConfigKeys, error)
-	pullAutomateConfigs() (map[string]*dc.AutomateConfig, error)
-	pullChefServerConfigs() (map[string]*dc.AutomateConfig, error)
-	fetchInfraConfig() (*ExistingInfraConfigToml, error)
-	generateInfraConfig() (*ExistingInfraConfigToml, error)
-	fetchAwsConfig() (*AwsConfigToml, error)
-	generateAwsConfig() (*AwsConfigToml, error)
+	pullOpensearchConfigs(removeUnreachableNodes bool) (map[string]*ConfigKeys, []string, error)
```

Comment:
Instead of passing `removeUnreachableNodes` as an argument, can we make it as a field in the `PullConfigImpl` to make the code much cleaner?